src/ConfigCatHooks.tsx

```diff
@@ -14,6 +14,7 @@ function useFeatureFlag<T extends SettingValue>(key: string, defaultValue: T, us
   const [loading, setLoading] = useState(true);
 
   useEffect(() => {
+    setLoading(true);

```

Comment:
I'd introduce a new state (e.g. evaluating/reevaluating) instead of setting the loading to true in this case. The loading state was introduced to indicate the config.json download in the background. I think this case a bit different. I think a new evaluating state could solve your problem. What do you think?

Reply:
hey,

I like for example react-query's approach
isLoading - fetching data, cache is empty
isFetching - fetching data, but we have something in cache

that would work for us, yes.

Reply:
Awesome. Could you please update the PR?

Reply:
will do. Can't promise any timeline though

Reply:
Hello @hakubo ,

Do you have any updates regarding this PR?

Cheers,